contracts/LiquidationManager.sol

```diff
@@ -0,0 +1,191 @@
+     *      Reverts with `InvalidTargetHealthFactor` if `_targetHealthFactor` is invalid.
+     */
+    constructor(
+        address comptroller_,
+        address accessControlManager_,
+        uint256 baseCloseFactorMantissa_,
+        uint256 defaultCloseFactorMantissa_,
+        uint256 targetHealthFactor_
+    ) {
+        ensureNonzeroAddress(comptroller_);
+        if (baseCloseFactorMantissa_ > mantissaOne || baseCloseFactorMantissa_ > defaultCloseFactorMantissa_) {
+            revert InvalidBaseCloseFactor();
+        }
+        if (defaultCloseFactorMantissa_ > mantissaOne) {
+            revert InvalidDefaultCloseFactor();
+        }
+        if (targetHealthFactor_ < mantissaOne) {
+            revert InvalidTargetHealthFactor();
+        }
+
+        baseCloseFactorMantissa = baseCloseFactorMantissa_;
+        defaultCloseFactorMantissa = defaultCloseFactorMantissa_;
+        targetHealthFactor = targetHealthFactor_;
+        comptroller = ComptrollerInterface(comptroller_);
+
+        __AccessControlled_init_unchained(accessControlManager_);
+    }
+
+    /**
+     * @notice Enables or disables the dynamic close factor for a specific market.
+     * @param market The address of the market to update.
+     * @param enabled Boolean indicating whether the dynamic close factor should be enabled or disabled.
+     */
+    function setDynamicCloseFactorEnabled(address market, bool enabled) external {
+        _checkAccessAllowed("setDynamicCloseFactorEnabled(address,bool)");
+
+        (bool isListed, , , , ) = comptroller.markets(market);
+        if (!isListed) {
+            revert MarketNotListed(market);
+        }
+        dynamicCloseFactorEnabled[market] = enabled;
+    }
+
+    /**
+     * @notice Enables or disables the dynamic liquidation incentive for a specific market.
+     * @param market The address of the market to update.
+     * @param enabled Boolean indicating whether the dynamic liquidation incentive should be enabled or disabled.
+     */
+    function setDynamicLiquidationIncentiveEnabled(address market, bool enabled) external {
+        _checkAccessAllowed("setDynamicLiquidationIncentiveEnabled(address,bool)");
+
+        (bool isListed, , , , ) = comptroller.markets(market);
+        if (!isListed) {
+            revert MarketNotListed(market);
+        }
+        dynamicLiquidationIncentiveEnabled[market] = enabled;
+    }
+
+    /**
+     * @notice Calculate the close factor for a liquidation
+     * @param borrowBalance The borrow balance of the borrower
+     * @param wtAvg The weighted average of the collateral
+     * @param totalCollateral The total collateral available for liquidation
+     * @param dynamicLiquidationIncentive The dynamic liquidation incentive, scaled by 1e18
+     * @param maxLiquidationIncentive The maximum liquidation incentive allowed
+     * @return closeFactor The calculated close factor, scaled by 1e18
+     */
+    function calculateCloseFactor(
```

Comment:
`calculateDynamicCloseFactor`? to be consistent with the "dynamic" liquidation incentive

Reply:
[f091f37](https://github.com/VenusProtocol/venus-protocol/pull/604/commits/f091f376f3fb2236ff929ed28f1228534b7f2ed3)